src/components/schedule-page/validation-drawer/validation-drawer.component.tsx

```diff
@@ -37,18 +37,11 @@ export default function ValidationDrawerComponent(): JSX.Element {
       </Button>
       <Drawer open={open} onClose={(): void => toggleDrawer(false)} anchor={"right"}>
         <SpanErrors errors={errors} />
-        <List>
-          {errors?.map(
-            (error, index): JSX.Element => (
-              <ListItem key={index + error.message.slice(0, 5)}>
-                <ListItemIcon>
-                  <ErrorOutlineIcon />
-                </ListItemIcon>
-                <ListItemText primary={error.message} />
-              </ListItem>
-            )
-          )}
-        </List>
+        {errors?.map(
+          (error): JSX.Element => (
+            <ErrorListItem error={error} />
```

Comment:
Brakuje klucza. 
W reakcie klcuze są bardzo ważne ponieważ za ich pomocą on wyznacza, czy komponent się zmienił i czy on musi być wyrenderowany od nowa czy nie. Teraz errory w tej liście co ty wygenerowałaś nie będą się zmieniać.
https://reactjs.org/docs/reconciliation.html#recursing-on-children

To wszystko jest dobrze opisane tutaj:
https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318

Reply:
Też myśle, że było by fajnie wynieść to do osobnego komponentu:
`ErrorList`
Wtedy my moglibyśmy w wygodny i czytelny sposób obsłużyć to, że nie ma błędów:
```
{erros.length() && erros.map(el => ....)}
{!errors.length() && "Nie znaleziono błędów"}
```
W sumie można to zrobić i tutaj. 
Ale mi się wydaje, że będzie czytelniej wynieść to do osobnego folderu